### ISSUE TITLE
Eas build not working

### ISSUE THREAD

**Issue by Jonatthu** [expo/eas-cli#729]
### Summary

When running
`eas build -p ios --profile=simulator --local`

### Environment

  EAS CLI 0.34.1 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.13.0 - /usr/local/bin/node
      Yarn: 1.22.17 - /usr/local/bin/yarn
      npm: 8.1.0 - /usr/local/bin/npm
    Utilities:
      Git: 2.31.0 - /opt/homebrew/bin//git
    npmPackages:
      expo-cli: ~4.12.11 => 4.12.11 
    npmGlobalPackages:
      eas-cli: 0.34.1
      expo-cli: 4.12.11

### Error output

[PREPARE_CREDENTIALS] Preparing credentials
[PREPARE_CREDENTIALS] Creating keychain
[PREPARE_CREDENTIALS] 
TypeError: Cannot read properties of undefined (reading 'data')
    at IosCredentialsManager.prepare (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/ios/credentials/manager.js:59:75)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/builders/ios.js:23:20
    at async BuildContext.runBuildPhase (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/context.js:44:28)
    at async Object.iosBuilder (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/builders/ios.js:22:29)
    at async Object.buildIosAsync (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/ios.js:43:27)
    at async Object.buildAsync (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/build.js:27:32)
    at async main (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/main.js:16:9)
[CLEAN_UP_CREDENTIALS] Destroying keychain - /var/folders/wr/sss6fkwj5l7b7l51g2kxktq80000gp/T/turtle-v2-8886ae28-2764-484c-afa5-258af612ff5e.keychain

Build failed
Cannot read properties of undefined (reading 'data')

### Reproducible demo or steps to reproduce from a blank project

This repo plus: 

eas.json

```
{
  "build": {
    "simulator": {
      "developmentClient": true,
      "ios": {
        "simulator": true
      }
    },
    "development": {
      "distribution": "internal",
      "android": {
        "buildType": "apk"
      }
    },
    "production": {
      "distribution": "store",
      "android": {
        "buildType": "app-bundle"
      }
    }
  },
  "cli": {
    "version": ">= 0.34.1"
  }
}
```

**Comment by dsokal** [comment]
Hi!
Try upgrading `eas-cli-local-build-plugin` to the latest version. (`yarn global add eas-cli-local-build-plugin` or `npm i -g eas-cli-local-build-plugin`)
Currently, in EAS CLI, we don't check if the version of the plugin is compatible with CLI. We'll fix it soon.

**Comment by Jonatthu** [comment]
Hello @dsokal this worked well, thanks!
But also when using "jsEngine": "hermes", on the app.json, this commands fails with error message:

```
[INSTALL_PODS] https://github.com/cocoapods/cocoapods/issues/new
[INSTALL_PODS] [!] `<PBXResourcesBuildPhase UUID=`13B07F8E1A680F5B00A75B9A`>` attempted to initialize an object with an unknown UUID. `E00711A40F3D448785C5B16C` for attribute: `files`. This can be the result of a merge and the unknown UUID is being discarded.
[INSTALL_PODS] 
Error: pod exited with non-zero code: 1
    at ChildProcess.completionListener (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/spawn-async/build/spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:510:26)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
    ...
    at Object.spawnAsync [as default] (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/spawn-async/build/spawnAsync.js:17:21)
    at Object.spawn [as default] (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/turtle-spawn/dist/index.js:17:42)
    at Object.installPods (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/ios/pod.js:12:33)
    at /usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/builders/ios.js:38:25
    at BuildContext.runBuildPhase (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/context.js:44:34)
    at Object.iosBuilder (/usr/local/lib/node_modules/eas-cli-local-build-plugin/node_modules/@expo/build-tools/dist/builders/ios.js:37:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.buildIosAsync (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/ios.js:43:27)
    at async Object.buildAsync (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/build.js:27:32)
    at async main (/usr/local/lib/node_modules/eas-cli-local-build-plugin/dist/main.js:16:9)

Build failed
pod exited with non-zero code: 1

```

**Comment by Jonatthu** [comment]
Also expo run:ios
```
Failed to build iOS project. "xcodebuild" exited with error code 65.
To view more error logs, try building the app with Xcode directly, by opening /Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/iTrustCapital.xcodeproj.

Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project /Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/iTrustCapital.xcodeproj -configuration Debug -scheme iTrustCapital -destination id=D96DEC46-E10E-44C5-B4DA-BB9F3685AE78

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

note: Using new build system
note: Planning
Analyze workspace

Create build description
Build description signature: 906aba72c2c3a221cede519607fbc328
Build description path: /Users/jonatthu/Library/Developer/Xcode/DerivedData/iTrustCapital-drpxquwmlcossxestqtrrhagzxod/Build/Intermediates.noindex/XCBuildData/906aba72c2c3a221cede519607fbc328-desc.xcbuild

note: Build preparation complete
note: Building targets in parallel
/Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/Pods/Target Support Files/Pods-iTrustCapital/Pods-iTrustCapital.debug.xcconfig:1:1: error: unable to open file (in target "iTrustCapital" in project "iTrustCapital")
/Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/Pods/Target Support Files/Pods-iTrustCapital/Pods-iTrustCapital.debug.xcconfig:1:1: error: unable to open file (in target "iTrustCapital" in project "iTrustCapital")
/Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/Pods/Target Support Files/Pods-iTrustCapital/Pods-iTrustCapital.debug.xcconfig:1:1: error: unable to open file (in target "iTrustCapital" in project "iTrustCapital")
/Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/ios/Pods/Target Support Files/Pods-iTrustCapital/Pods-iTrustCapital.debug.xcconfig:1:1: error: unable to open file (in target "iTrustCapital" in project "iTrustCapital")


** BUILD FAILED **

Build logs written to /Users/jonatthu/Documents/ITrustCapital.Application/frontend/apps/portfolio/.expo/xcodebuild.md
```

**Comment by brentvatne** [comment]
hey @Jonatthu! if you can please share a reproducible example of your issue we would be happy to help. also, please create a new issue for the specific problem that you're encountering
